### PR TITLE
feat: add support for signup user email verification

### LINKED ISSUE #536: Add support for signup user email verification
For Backend.AI public service, let's add a feature to support email verification.

### PR DESCRIPTION
**Pull request by adrysn**
This PR resolves #536.

* New component: `backend-ai-email-verification-view`.
* Add route `verify-email` for email verification.
  - Visiting this url will try to verify token, not login.
  - Token should be delivered as a query string by `verification_code`.
    - If verified, success dialog is popup and redirects to root page.
    - If not verified, fail dialog is popped-up with re-send verification email form.
  - If token is not supplied, fail dialog is popped-up with re-send verification email form.
* New `Cloud` class in the JavaScript SDK.
  - Add `verify_email` and `send_verification_email` in the `Cloud` API.

## How to test signup with email verification feature
* `cloud-beta-plugin` should be installed (by pip) in the manager.
* Signup with test account (with invitation token). You will see a dialog which states that an email is sent. Check your mailbox to discover delivered email.
  image
* Click VERIFY link. You should see email verified dialog. Or, you can see fail dialog if token is expired, or some error occurs.
  image
* In case of failure, user can resend a verification email again in the fail dialog.

### CODE REVIEW

File: src/components/backend-ai-email-verification-view.ts [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+          width: 100%;
+        }
+      `
+    ];
+  }
+
+  _initClient(apiEndpoint: string) {
+    this.consoleShell = document.querySelector('#console-shell');
+    this.consoleShell.appBody.style.visibility = 'visible';
+    this.notification = this.consoleShell.notification;
```
Comment (inureyes) [inline src/components/backend-ai-email-verification-view.ts]: This should be `this.notification = globalThis.lablupNotification;`

File: src/components/backend-ai-email-verification-view.ts [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+        mwc-textfield {
+          width: 100%;
+        }
+      `
+    ];
+  }
+
+  _initClient(apiEndpoint: string) {
+    this.consoleShell = document.querySelector('#console-shell');
+    this.consoleShell.appBody.style.visibility = 'visible';
```
Comment (inureyes) [inline src/components/backend-ai-email-verification-view.ts:49]: Why do we need this part? This part produces FOUC issue occasionally.

File: src/components/backend-ai-console.ts [hunk 3]
```diff
@@ -493,6 +500,10 @@ export default class BackendAIConsole extends connect(store)(LitElement) {
         this.menuTitle = _text("console.menu.Logs");
         this.updateTitleColor('var(--paper-deep-orange-800)', '#efefef');
         break;
+      case 'verify-email':
```
Comment (inureyes) [inline src/components/backend-ai-console.ts]: Let's use summary and verify-email block together.

File: src/components/backend-ai-email-verification-view.ts (hunk 1 above)
Comment (adrysn) [reply src/components/backend-ai-email-verification-view.ts]: Fixed.

File: src/components/backend-ai-email-verification-view.ts (hunk 2 above)
Comment (adrysn) [reply src/components/backend-ai-email-verification-view.ts:49]: Without this, verification dialogs are not visible to users. Is there other workaround to make dialogs visible?

File: src/components/backend-ai-console.ts (hunk 3 above)
Comment (adrysn) [reply src/components/backend-ai-console.ts]: Merged `verify-email` block with `summary`.

File: src/components/backend-ai-email-verification-view.ts (hunk 2 above)
Comment (inureyes) [reply src/components/backend-ai-email-verification-view.ts:49]: Ok. It is only called while email verification is processed. Let's see after merging.

**Pull request merged**